### DISCUSSION TITLE
BRouter (Beta)

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#15] [Show and tell]
BRouter 1.7 is available:

https://play.google.com/store/apps/details?id=btools.routingapp

Release notes: https://github.com/abrensch/brouter/releases/tag/v1.7.0

BRouter 1.7 offers the "**Straight line**" option in route planning.

**Comment by Cer-o** [comment]
I have version 1.6.3 installed to use it with Kurviger.
Is it also going to be used by Cruiser? Is the version I have valid or should I install the one you shared?

**Comment by devemux86** [comment]
Cruiser can use both BRouter versions and enables the new features when BRouter (Beta) is installed.

BRouter (Beta) is for testing its [new features](https://github.com/abrensch/brouter/pull/441), like "Straight line" profile, u-turn improvements, etc.

**Reply by DolomitesTom** [reply]
How long do not beta users usually have to wait until it is available via play store?

**Reply by devemux86** [reply]
It depends on the developers of each app when they make the release.

BRouter (Beta) apk is kindly provided by BRouter developers for testing.

**Reply by WolfgangZ42** [reply]
> BRouter (Beta) is for testing its https://github.com/abrensch/brouter/pull/441, like "Straight line" profile, u-turn improvements, etc.

Today I tested the "u-turn improvements":
* put assign correctMisplacedViaPoints = 1 and assign correctMisplacedViaPointsDistance = 100 into .brf file
* designed a tour with misplaced shaping points in kurviger.de 
* exported it as .kurviger and as .gpx file
* drove .kurviger with Cruiser - all 4 u-turns are announced 
* same with .gpx - the u-turns are perfectly eliminated 

Even it's not really important (because I look after u-turns before exporting a .kurviger file) it's interesting to see that BRouter is ahead here.

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the feedback and the tests!

cc @afischerdev

**Comment by devemux86** [comment]
A new version of BRouter (Beta) is available: https://github.com/devemux86/cruiser/discussions/15#discussion-4757829

**Reply by WolfgangZ42** [reply]
If 1.6.4x is at my smartphone in charge and 1.6.4y is now installed will all my profiles and maps still used? Or is a total re-establishing of BRouter necessary? Thanks for a hint.

**Reply by devemux86** [reply]
Android apk can be installed on top of the existing installation, it works like the Google Play updates.

**Comment by devemux86** [comment]
A new version of BRouter (Beta) is available: https://github.com/devemux86/cruiser/discussions/15#discussion-4757829

This is needed with [Cruiser 3.0.11](https://play.google.com/store/apps/details?id=gr.talent.cruiser), if you use BRouter (Beta).

**Comment by devemux86** [comment]
A new version of BRouter (Beta) is available: https://github.com/devemux86/cruiser/discussions/15#discussion-4757829

**Comment by devemux86** [comment]
A new version of BRouter (Beta) is available: https://github.com/devemux86/cruiser/discussions/15#discussion-4757829

**Comment by devemux86** [comment]
> can somebody ask the Brouter developer to create a curvy route option :-) ?

Searching [BRouter](https://github.com/abrensch/brouter/issues), we may be pleasantly surprised in the future:
- https://github.com/abrensch/brouter/issues/460
- https://github.com/abrensch/brouter/issues/258
- https://github.com/abrensch/brouter/issues/476
- https://github.com/abrensch/brouter/issues/486

**Comment by DolomitesTom** [comment]
Thanks, don't want to step too much into details, but it sounds useful, especially the "consider surroundings" sounds like quicker calculate detours if not the pre-planned route is taken.

**Comment by devemux86** [comment]
A new version of BRouter (Beta) is available: https://github.com/devemux86/cruiser/discussions/15#discussion-4757829

**Comment by DolomitesTom** [comment]
The usage of BRouter Beta via Google App Store is not possible? I have forbidden installation from external sources on my device.

**Reply by devemux86** [reply]
This is how @afischerdev provides BRouter (Beta) apk, probably for faster releases.

**Comment by ToToGSARider** [comment]
Was ist eurer Meinung nach die bessere Wahl für das Offline Routing? Ich habe das früher sowieso in der kurviger 2 App so gemacht, dass ich die Neuberechnung deaktiviert hatte. Was soll ich einstellen? In K2 hatte ich Online kurviger und Offline BRouter. Diese Einstellung gibt es hier ja nicht. Wenn ich jetzt zum Beispiel Offline GraphHopper einstellen würde und die Routenneuberechnung deaktiviert lasse, dann bleibt meine kurviger Route doch unberührt?

What do you think is the better choice for offline routing? I used to do this anyway in the curvy 2 app so I had recalculation disabled. What should I set? In K2 I had online curviger and offline BRouter. This setting does not exist here. If I would now set for example Offline GraphHopper and leave the route recalculation disabled, then my curviger route remains unaffected?

**Reply by devemux86** [reply]
@ToToGSARider 

Cruiser offers 2 offline routers:
- BRouter (offline) with powerful customizable routing profiles and cross border routing
- GraphHopper (offline) with street names / speed limits (only 1 country can be loaded)

Cruiser re-routing only calculates the path from the user location to the next waypoint, the rest of the **route remains unchanged**.

---

Cruiser bietet 2 Offline-Router:
- BRouter (offline) mit leistungsstarken anpassbaren Routing-Profilen und grenzüberschreitendem Routing
- GraphHopper (offline) mit Straßennamen / Geschwindigkeitsbegrenzungen (es kann nur 1 Land geladen werden)

Cruiser Re-Routing berechnet nur den Weg vom Benutzerstandort zum nächsten Wegpunkt, der Rest der **Route bleibt unverändert**.

**Comment by ToToGSARider** [comment]
Okay, verwendet der Cruiser gegebenenfalls meinen installierten BRouter? Ich habe eine Version wo die letzte Aktualisierung am 24.12.2021 war? Ich habe das Programm damals einmal installiert und wenn nur die automatischen Updates vom Playstore erhalten.

Okay, does the Cruiser use my installed BRouter if necessary? I have a version where the last update was 24.12.2021? I installed the program once at that time and if only get the automatic updates from the playstore.

**Reply by devemux86** [reply]
@ToToGSARider 

Cruiser uses the same installed BRouter app.
But you should open BRouter app from time to time and update is routing data.

See the images here:
- https://github.com/devemux86/cruiser/discussions/4#discussioncomment-4533509

---

Cruiser verwendet die gleiche installierte BRouter-App.
Sie sollten jedoch die BRouter-App von Zeit zu Zeit öffnen und die Routing-Daten aktualisieren.

Sehen Sie sich die Bilder hier an:
- https://github.com/devemux86/cruiser/discussions/4#discussioncomment-4533509

**Comment by devemux86** [comment]
A new version of BRouter (Beta) is available: https://github.com/devemux86/cruiser/discussions/15#discussion-4757829

**Reply by WolfgangZ42** [reply]
Thanks Emux for keep us informed! One question: could we look anywhere (you'll know where) after the change list of this version? To get motivation for installing it ... 😉

**Reply by devemux86** [reply]
As I was informed by @afischerdev this a a 'nearly' final version, probably the next update on Google Play.
Please everyone test it well! :slightly_smiling_face: 

---

> could we look anywhere (you'll know where) after the change list of this version?

@afischerdev can answer better, but 1.6.4 could be summarized here:
- https://github.com/abrensch/brouter/pull/441

BRouter revision history is here: https://brouter.de/brouter/revisions.html

**Comment by devemux86** [comment]
A new version of BRouter (Beta) is available: https://github.com/devemux86/cruiser/discussions/15#discussion-4757829

**Comment by devemux86** [comment]
A new version of BRouter (Beta) is available: https://github.com/devemux86/cruiser/discussions/15#discussion-4757829

**Comment by devemux86** [comment]
A new version of BRouter (Beta) is available: https://github.com/devemux86/cruiser/discussions/15#discussion-4757829

**Comment by devemux86** [comment]
A new version of BRouter (Beta) is available: https://github.com/devemux86/cruiser/discussions/15#discussion-4757829

**Comment by devemux86** [comment]
Many interesting things are planned in BRouter (offline)! :+1: 

Copying from [here](https://github.com/abrensch/brouter/pull/556):

_"forest to find nice ways in or near forest
river to find ways along water
noise to avoid bigger streets
town to avoid bigger cities
traffic was used before but is now generated by database call
...
This needs also an update for the profiles to make use of the new tags."_

**Reply by DolomitesTom** [reply]
Sounds great

**Reply by Ciutnik** [reply]
We'll see how it works in practice, but it looks nice. If BRouter returned street names it would be very nice! :-)

**Comment by devemux86** [comment]
More on BRouter's exciting progress with new profile options, see below with screenshots:

- https://github.com/abrensch/brouter/issues/486#issuecomment-1574827962

Consider traffic, wet, unpaved, steepness, elevation, forests, rivers, noise, towns, ... :+1:

**Reply by Ciutnik** [reply]
Could this be taken as an announcement of new avoidance/preference options in Cruiser, such as "avoid cities" or "prefer forests"? Not bad 👍

By the way, I know you're in contact with a BRouter developer who reads this forum (but I don't dare mention him here...). Is there any chance of introducing support for street names and speed limits in BRouter in the future? It is often a "good to have" option, but in some cases (e.g. pedestrian or bicycle city navigation "by ear") it is crucial. I know there have been requests like this in the past, but to no avail. I don't want to be pushy, but maybe we should start a some petition? I would sign with both hands and feet 🙏

**Reply by MotoUKRider** [reply]
If these considerations are available to apps like Cruiser as a toggle on/off, it has the potential to allow for some very good vehicle profiles. When I relied on CoPilot (rely meaning good app but zero support/pro-active dev), I used the "Car" profile for fast A-to-B routing and "Motorcycle" for a more adventurous route away from big roads (you could set a level for each road type: Strongly Favor, Neutral, Avoid etc)

Road-Names and Speed-Limits in BRouter would be handy - although the GraphHopper speed display is not always available I found, and the metric/imperial conversion inside Cruiser probably needs a "Format" routine - we do *not* have 59mph speed-limits in the UK :)

**Reply by devemux86** [reply]
@MotoUKRider 

> the metric/imperial conversion inside Cruiser probably needs a "Format" routine - we do not have 59mph speed-limits in the UK :)

Can you please create a new topic with a specific example?

Max speeds come from the routers and they use Metric units (km/h or m/s), so apps must make the conversion.
